### PR TITLE
Beyondr-20240916

### LINKED ISSUE #134: [UI] SVG icons
* Logo should use svg icon
* Footer icon should use SVG

### LINKED ISSUE #185: [FE] reduce the number of different grey backgrounds
--> Let us use the colors from the guidelines (Product UI colors):
Image

The background here is e5e7eb
Image

The card background here is e8e8e8
Image

The grey background here is f1f5f9
Image

### LINKED ISSUE #187: [FE] Increase brand mark
Lets increase the logo, such that the brand mark is as high as the button on the right

Image

### LINKED ISSUE #188: show error "image not available"
If an image is not available, let us show a horizontally and vertically centered text "Image not available"

Instead of this

Image

### LINKED ISSUE #190: [create] enforce character limits
I just minted a hypercert with a very looooong title :D https://testnet.hypercerts.org/hypercerts/11155111-0xa16DFb32Eb140a6f3F2AC68f41dAd8c7e83C4941-213697326426349355054999253467150436794368

Let's enforce some character limits
- Title: Max. 100 characters
- Description: Max. 5000 characters
- Single work tags: 50 characters
- Single contributor: 50 characters

If it is easy we can also limit the number of work tags to 20.

edit: changed the description limit to 5000 based on comments below

### PR DESCRIPTION
**Pull request by thebeyondr**
## Key changes
- Refactored UI components for improved consistency and aesthetics
- Updated color scheme to use new accent color
- Replaced PNG logo with SVG version
- Removed unused `hypercert-mini-display` component
- Updated dependencies

### Commits
- [Refactor window](https://github.com/hypercerts-org/hypercerts-app/commit/d8fd307e23824fb8e248e6b9b67790d4e55b3609)
- [Remove unused component](https://github.com/hypercerts-org/hypercerts-app/commit/5487c96ba2f3bf7e1c09936315098a058b715cd4)
- [Clean up comment](https://github.com/hypercerts-org/hypercerts-app/commit/2998caad9e852f69b1b01272e9eea8e7ca275452)
- [Update form character limits](https://github.com/hypercerts-org/hypercerts-app/commit/f657870fbb730097046c972bd0be8c7ff3677246)
- [Add wait before card snapshot, add card preview (dev)](https://github.com/hypercerts-org/hypercerts-app/commit/4b8124f44ae81bb343ae467b9c68817a1ef94a77)
- [Update title and scope styles](https://github.com/hypercerts-org/hypercerts-app/commit/03bcc3d58f00b1b96c871aa0ec257a68d183da30)
- [Tweak scope styles](https://github.com/hypercerts-org/hypercerts-app/commit/419f1f212995675499713dbdfdd3acdcae9f8fd5)
- [Update hypercert card layout](https://github.com/hypercerts-org/hypercerts-app/commit/daf0d91d46014b2cd4ec12ed87b8621c79cee8ae)
- [Update card title sizing](https://github.com/hypercerts-org/hypercerts-app/commit/b8476239a84fe888656445cae069eccd59c0171f)
- [Format code and update lock file](https://github.com/hypercerts-org/hypercerts-app/commit/db84735ef32d9ad9173a242b5293cc80201cd393)
- [Use dynamic import for tagify](https://github.com/hypercerts-org/hypercerts-app/commit/beb6c4d18dc618c4f1360ade4abb50620dd34541)
- [Update placeholder image if none is found, update logic for image route](https://github.com/hypercerts-org/hypercerts-app/commit/b53be7a6b3e45fc0fb9a7990accd03fa3302febc)
- [Increase logo size in navbar](https://github.com/hypercerts-org/hypercerts-app/commit/439610d5a198bcc33cbfa263ff65f4e84a55053e)
- [Update hypercert logo to svg](https://github.com/hypercerts-org/hypercerts-app/commit/a430f09c1f62e4b974306dd9e8948f3b7b427d68)
- [Update explore styles](https://github.com/hypercerts-org/hypercerts-app/commit/4311b55704f319dd6501dde44a486edaa2a087d5)
- [Update hypercert-window styling](https://github.com/hypercerts-org/hypercerts-app/commit/b526be22569b88198042fe7226e5d8e5f6ac86dc)
- [Update UI gray colors](https://github.com/hypercerts-org/hypercerts-app/commit/31ea23b6e2216d8f63b1769bebfb6559bbd10a26)

### New files
- `/public/hypercerts-logo.svg`

### Renamed files
None

### Deleted files
- `/components/hypercert/hypercert-mini-display.tsx`
- `/public/hypercerts-logo.png`

### Dependencies
- Updated `@types/react` from 18.3.3 to 18.3.8
- Updated `zustand` from 4.5.3 to 4.5.5
- Updated `use-sync-external-store` from 1.2.0 to 1.2.2

Closes #190, Closes #134, Closes #187, Closes #188, Closes #185

### CODE REVIEW

File: app/api/hypercerts/[hypercertId]/image/route.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export async function GET(
     const imageOrUrl = res.hypercerts.data?.[0]?.metadata?.image;
 
     // Use placeholder image if no image URL or data is found
-    if (!imageOrUrl) {
+    if (!imageOrUrl || imageOrUrl === "https://hypercerts.org/logo.png") {
```
Comment (0xRowdy) [inline app/api/hypercerts/[hypercertId]/image/route.ts:81]: Should this be the SVG?

File: app/globals.css [hunk 2]
```diff
@@ -22,7 +22,8 @@
     --muted: 210 40% 96.1%;
     --muted-foreground: 215.4 16.3% 46.9%;
 
-    --accent: 210 40% 96.1%;
+    /* --accent: 210 40% 96.1%; */
```
Comment (0xRowdy) [inline app/globals.css:25]:
```suggestion
```

File: app/hypercerts/new/form-steps.tsx [hunk 3]
```diff
@@ -511,6 +513,19 @@ const FormSteps = ({
...
+          <div className="relative w-[336px] h-[420px] mx-auto">
+            <Image
+              src={form.watch("cardImage")}
+              alt="Hypercert Card Preview"
+              layout="fill"
+              objectFit="contain"
+            />
+          </div>
+        </div>
+      )} */}
```
Comment (0xRowdy) [inline app/hypercerts/new/form-steps.tsx:528]:
```suggestion
```

File: components/hypercert/evaluate-drawer.tsx [hunk 4]
```diff
@@ -67,6 +66,13 @@ export function EvaluateDrawer({ hypercert }: { hypercert: HypercertFull }) {
     });
   const [comments, setComments] = useState<string>("");
   const [uid, setUid] = useState<string>();
+  const [Tags, setTags] = useState<any>(null);
```
Comment (0xRowdy) [inline components/hypercert/evaluate-drawer.tsx:69]: Can this be an array of strings or null?
```suggestion
  const [Tags, setTags] = useState<string[] | null>(null);
```

File: components/hypercert/hypercert-window.tsx [hunk 5]
```diff
@@ -59,8 +59,8 @@ const HypercertWindow = ({
 
   return (
     <Link href={`/hypercerts/${hypercertId}`}>
-      <article className="transition-transform duration-300 hover:-translate-y-2 relative group bg-black/10 rounded-lg overflow-hidden">
-        <div className="h-[320px] w-full relative p-1">
+      <article className="transition-transform duration-300 hover:-translate-y-2 relative group bg-accent rounded-lg overflow-hidden pt-3">
+        <div className="h-[260px] w-full relative p-1">
```
Comment (0xRowdy) [inline components/hypercert/hypercert-window.tsx:63]: Should we be using rem?
```suggestion
        <div className="h-[16.25rem] w-full relative p-1">
```

**Review by 0xRowdy — APPROVED** [review]
I left a few comments and questions but lgtm

**Comment by bitbeckers** [comment]
Looking great @thebeyondr ! @holkexyz could you review the test deployment?

One thing @thebeyondr is that the images on the page below are full width, iirc there should be a padding on the x-axis?

https://hypercerts-n16ugk42l-hypercerts-foundation.vercel.app/explore?p=53

Screenshot 2024-09-23 at 16 49 35

File: app/api/hypercerts/[hypercertId]/image/route.ts (hunk 1 above)
Comment (thebeyondr) [reply app/api/hypercerts/[hypercertId]/image/route.ts:81]: no, its seems the api returns this logo when an image is not found for the hypercert, however we want to show the svg from our static files instead.

File: app/hypercerts/new/form-steps.tsx (hunk 3 above)
Comment (thebeyondr) [reply app/hypercerts/new/form-steps.tsx:528]: this was a card preview so I can see the final snapshot of the card without having to mint it

File: components/hypercert/evaluate-drawer.tsx (hunk 4 above)
Comment (thebeyondr) [reply components/hypercert/evaluate-drawer.tsx:69]: Its actually a placeholder for a dynamic import of a Tag component, not a list of tags (see useEffect on line 70)

File: components/hypercert/hypercert-window.tsx (hunk 5 above)
Comment (thebeyondr) [reply components/hypercert/hypercert-window.tsx:63]: I'm for use of rems, however afaik rems are useful when we want things to scale based on some root size, like font-size. My thinking was the cards should be 260px tall no matter what... but I could go either way - just more work to calculate.

**Comment by thebeyondr** [comment]
> Looking great @thebeyondr ! @holkexyz could you review the test deployment?
> 
> One thing @thebeyondr is that the images on the page below are full width, iirc there should be a padding on the x-axis?
> 
> https://hypercerts-n16ugk42l-hypercerts-foundation.vercel.app/explore?p=53
> 
> Screenshot 2024-09-23 at 16 49 35

added padding now, don't remember if there was before. We don't really encounter this issue since the cards are more vertical now anyways

**Review by 0xRowdy — APPROVED** [review]

File: app/hypercerts/new/form-steps.tsx (hunk 3 above)
Comment (bitbeckers) [reply app/hypercerts/new/form-steps.tsx:528]: But... that's a useful feature no? Can't that just always be displayed on the last step? No need to add extra work for this, buit if it's already there?

**Pull request merged**